src/main/scala/is/hail/utils/IntervalTree.scala

```diff
@@ -1,99 +1,90 @@
 package is.hail.utils
 
+import is.hail.annotations.ExtendedOrdering
 import is.hail.check._
 import org.json4s.JValue
 import org.json4s.JsonAST.JObject
 
 import scala.collection.mutable
 import scala.language.implicitConversions
-import scala.math.Ordering.Implicits._
 import scala.reflect.ClassTag
 
 // interval inclusive of start, exclusive of end: [start, end)
-case class Interval[T](start: T, end: T)(implicit ev: Ordering[T]) extends Serializable {
-  import ev._
+case class Interval(start: Any, end: Any) extends Serializable {
+  def contains(pord: ExtendedOrdering, position: Any): Boolean = pord.gteq(position, start) && pord.lt(position, end)
```

Comment:
What happens if you create an Interval where start and end are two different types? Is that tested for anywhere?

Reply:
That's not allowed, the same way creating an `IndexedSeq[Any]("foo", 5)` is not allowed to represent a value of type `Array[Int32]`.  It is tested by TInterval._typeCheck, which verifies the start and end both are instances of `pointType`.

```
  def _typeCheck(a: Any): Boolean = a.isInstanceOf[Interval] && {
    val i = a.asInstanceOf[Interval]
    pointType.typeCheck(i.start) && pointType.typeCheck(i.end)
  }
```